.github/workflows/ci.yml

```diff
@@ -28,24 +28,28 @@ jobs:
         # Ignore the failure of a step and avoid terminating the job.
         continue-on-error: true
         with:
-          key: docker-cache-${{ hashFiles('.git/modules/prysm/*/HEAD', '.git/modules/geth/*/HEAD') }}
+          key: docker-cache-${{ hashFiles('.git/modules/prysm/*/HEAD', '.git/modules/geth/*/HEAD', './shared/Dockerfile.shared') }}
+
+      # build images as a separate step to better utilize caching
+      # TODO: change to pure "docker-compose build" once we can successfully build lighthouse and lodster
+      - run: docker-compose build prysm-beacon-node prysm-beacon-node-follower prysm-validator-node build-shared-states execution-node execution-node-2
```

Comment:
This saves lots of time by caching containers, each of the following tests only took 5 min to run now